### PR TITLE
backup: initial support backup

### PR DESCRIPTION
**Pull request by overvenus**
This PR adds basic proto to support backup. Restore is not included and will be proposed separately.

### CODE REVIEW

File: proto/backup.proto [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    bytes start_key = 3;
+    bytes end_key = 4;
+    uint64 start_version = 5;
+    uint64 end_version = 6;
+}
+
+message Schema {
+    bytes db = 1;
+    bytes table = 2;
```
Comment (zhangjinpeng1987) [inline proto/backup.proto]: Is table contain how to create the table?

File: proto/backup.proto [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+message ClusterIDError {
```
Comment (zhangjinpeng1987) [inline proto/backup.proto]: Should contain err message.

**Comment by zhangjinpeng1987** [comment]
@BusyJay PTAL

File: proto/backup.proto (hunk 1 above)
Comment (overvenus) [reply proto/backup.proto]: Yes, it contains model.tableinfo. https://github.com/pingcap/parser/blob/8d44c03502f6b50e6111101c5504488d17f411b3/model/model.go#L194

File: proto/backup.proto (hunk 2 above)
Comment (overvenus) [reply proto/backup.proto]: The error message is included `Error`.

File: proto/backup.proto [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+message BackupRequest {
+    uint64 cluster_id = 1;
+
+    bytes start_key = 2;
+    bytes end_key = 3;
+    uint64 start_version = 4;
+    uint64 end_version = 5;
+
+    string path = 6;
```
Comment (zhangjinpeng1987) [inline proto/backup.proto]: Can this be remote url?

Comment (overvenus) [reply proto/backup.proto]: Yes, this is a URL.

**Pull request merged**